addons/containerd/template/base/install.sh

```diff
@@ -1,5 +1,25 @@
 
 CONTAINERD_NEEDS_RESTART=0
+CONTAINERD_DID_MIGRATE_FROM_DOCKER=0
+
+function containerd_pre_init() {
+    local src="$DIR/addons/containerd/$CONTAINERD_VERSION"
+
+    # Explicitly configure kubelet to use containerd instead of detecting dockershim socket
+    if [ -d "$DIR/kustomize/kubeadm/init-patches" ]; then
```

Comment:
what happens if there's not an init-patches directory yet?

Should this create it?

Reply:
I'm only copying this code but my hunch is that it is a way to detect if this is kubeadm addon init and not join